### PR TITLE
[DOCFIX] Improve software requirement doc

### PR DESCRIPTION
**Pull request by jiacheliu3**
The table of contents is missing.

Also there's a port on job worker not mentioned.

### CODE REVIEW

File: docs/en/deploy/Requirements.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ group: Install Alluxio
 priority: 9
 ---
 
+* Table of Contents
```
Comment (jiacheliu3) [inline docs/en/deploy/Requirements.md:9]: @Xenorith not sure how this is implemented, but i'm hoping to fix the missing toc on this doc

Comment (Xenorith) [reply docs/en/deploy/Requirements.md:9]: next time you can try it out yourself by running `jekyll serve` from `docs/` and view generated docs at localhost:4000

**Comment by jiacheliu3** [comment]
@Xenorith Good call! This is verified to work with `jekyll serve`.

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Comment by ZacBlanco** [comment]
The job worker data port is deprecated. We should remove it from the code. It is not used anywhere

**Pull request merged**